.ci/infrastructure/main.tf

```diff
@@ -0,0 +1,365 @@
+  }
+
+  data = {
+    "github_app_id"              = data.google_secret_manager_secret_version.github_app_id.secret_data
+    "github_app_installation_id" = data.google_secret_manager_secret_version.github_app_installation_id.secret_data
+    "github_app_private_key"     = data.google_secret_manager_secret_version.github_app_private_key.secret_data
+  }
+
+  type = "Opaque"
+}
+
+resource "kubernetes_namespace" "llvm_premerge_windows_runners" {
+  metadata {
+    name = "llvm-premerge-windows-runners"
+  }
+}
+
+resource "kubernetes_secret" "windows_github_pat" {
+  metadata {
+    name      = "github-token"
+    namespace = "llvm-premerge-windows-runners"
+  }
+
+  data = {
+    "github_app_id"              = data.google_secret_manager_secret_version.github_app_id.secret_data
+    "github_app_installation_id" = data.google_secret_manager_secret_version.github_app_installation_id.secret_data
+    "github_app_private_key"     = data.google_secret_manager_secret_version.github_app_private_key.secret_data
+  }
+
+  type = "Opaque"
+}
+
+
+resource "kubernetes_config_map" "linux_container_pod_template" {
+  metadata {
+    name      = "linux-container-pod-template"
+    namespace = "llvm-premerge-linux-runners"
+  }
+
+  data = {
+    "linux-container-pod-template.yaml" : "${file("linux_container_pod_template.yaml")}"
+  }
+}
+
+resource "helm_release" "github_actions_runner_controller" {
+  name       = "llvm-premerge-controller"
+  namespace  = "llvm-premerge-controller"
+  repository = "oci://ghcr.io/actions/actions-runner-controller-charts"
+  version    = "0.9.3"
+  chart      = "gha-runner-scale-set-controller"
+
+  depends_on = [
+    kubernetes_namespace.llvm_premerge_controller
+  ]
+}
+
+resource "helm_release" "github_actions_runner_set_linux" {
+  name       = "llvm-premerge-linux-runners"
+  namespace  = "llvm-premerge-linux-runners"
+  repository = "oci://ghcr.io/actions/actions-runner-controller-charts"
+  version    = "0.9.3"
+  chart      = "gha-runner-scale-set"
+
+  values = [
+    "${file("linux_runners_values.yaml")}"
+  ]
+
+  depends_on = [
+    kubernetes_namespace.llvm_premerge_linux_runners,
+    kubernetes_config_map.linux_container_pod_template,
+    kubernetes_secret.linux_github_pat
+  ]
+}
+
+resource "helm_release" "github_actions_runner_set_windows" {
+  name       = "llvm-premerge-windows-runners"
+  namespace  = "llvm-premerge-windows-runners"
+  repository = "oci://ghcr.io/actions/actions-runner-controller-charts"
+  version    = "0.9.3"
+  chart      = "gha-runner-scale-set"
+
+  values = [
+    "${file("windows_runner_values.yaml")}"
+  ]
+
+  depends_on = [
+    kubernetes_namespace.llvm_premerge_windows_runners,
+    kubernetes_secret.windows_github_pat
+  ]
+}
+
+resource "helm_release" "grafana-k8s-monitoring" {
+  name             = "grafana-k8s-monitoring"
+  repository       = "https://grafana.github.io/helm-charts"
+  chart            = "k8s-monitoring"
+  namespace        = var.namespace
```

Comment:
Why do you put grafana in the default namespace, instead of its own namespace, like you do for llvm-premerge-metrics?

Reply:
Good point. Switched it up to use the new `grafana` namespace.